### PR TITLE
Typescript 3.7

### PR DESCRIPTION
**Pull request by alexduf**
I tried to catch all the obvious uses of `variable || "defaultValue"` and of quasi false value in conditions, but I'm sure it's not exhaustive

### CODE REVIEW

File: typescript/src/link/deleteLink.ts [hunk 1]
```diff
@@ -24,17 +24,14 @@ async function deleteUserSubscription(subscriptionId: string): Promise<number> {
...
         return dynamoEvent.eventName == "REMOVE" &&
-            dynamoEvent.userIdentity &&
-            dynamoEvent.userIdentity.type == "Service" &&
-            dynamoEvent.userIdentity.principalId == "dynamodb.amazonaws.com" &&
-            dynamoEvent.dynamodb &&
-            dynamoEvent.dynamodb.Keys &&
-            dynamoEvent.dynamodb.Keys.subscriptionId
+            dynamoEvent.userIdentity?.type == "Service" &&
+            dynamoEvent.userIdentity?.principalId == "dynamodb.amazonaws.com" &&
+            dynamoEvent.dynamodb?.Keys?.subscriptionId
```
Comment (JamieB-gu) [inline typescript/src/link/deleteLink.ts]: Worth converting `==` to `===`?

File: typescript/src/subscription-status/googleSubStatus.ts [hunk 2]
```diff
@@ -22,11 +22,11 @@ interface SubscriptionStatusResponse {
 }
 
 function getPurchaseToken(headers: HttpRequestHeaders): string {
-    return headers["Play-Purchase-Token"] || headers["play-purchase-token"]
+    return headers["Play-Purchase-Token"] ?? headers["play-purchase-token"]
```
Comment (JamieB-gu) [inline typescript/src/subscription-status/googleSubStatus.ts]: What happens if `headers["play-purchase-token"]` is missing?

File: typescript/src/subscription-status/googleSubStatus.ts [hunk 3]
```diff
@@ -22,11 +22,11 @@ interface SubscriptionStatusResponse {
...
 }
 
 function googlePackageName(headers: HttpRequestHeaders): string {
-    const packageNameFromHeaders = headers["Package-Name"] || headers["package-name"];
+    const packageNameFromHeaders = headers["Package-Name"] ?? headers["package-name"];
```
Comment (JamieB-gu) [inline typescript/src/subscription-status/googleSubStatus.ts]: As above ^

File: typescript/src/subscription-status/googleSubStatus.ts (hunk 2 above)
Comment (alexduf) [reply typescript/src/subscription-status/googleSubStatus.ts]: The function returns null, and the server returns an HTTP 400 invalid request (see line 77)

File: typescript/src/subscription-status/googleSubStatus.ts (hunk 3 above)
Comment (alexduf) [reply typescript/src/subscription-status/googleSubStatus.ts]: line 33, we test for nullity of the value

File: typescript/src/subscription-status/googleSubStatus.ts (hunk 2 above)
Comment (JamieB-gu) [reply typescript/src/subscription-status/googleSubStatus.ts]: If the function returns `null`, should that be reflected in the return type?

**Pull request merged**